### PR TITLE
TNO-780, 785, 782, 787

### PR DESCRIPTION
**Pull request by jdtoombs**
Includes:
- npm package upgrade -> 0.0.202
- new content form toolbar
- new content form
- styling updates
- copy to clipboard for source url
- new navigate link

To note:
There are future tickets for toning, tags, time log. The layout has been updated to match these tickets but the styling will still need to be completed with those tickets.

Also Alessia's work will be added once merged.

image

Zoomed out view to capture most of the visuals:
image

### CODE REVIEW

File: app/editor/src/features/content/form/ContentForm.tsx [hunk 1]
```diff
@@ -100,8 +87,8 @@ export const ContentForm: React.FC<IContentFormProps> = ({
 
   const userId = userInfo?.id ?? '';
   const indexPosition = !!id ? page?.items.findIndex((c) => c.id === +id) ?? -1 : -1;
-  const enablePrev = indexPosition > 0;
-  const enableNext = indexPosition < (page?.items.length ?? 0) - 1;
+  // const enablePrev = indexPosition > 0;
```
Comment (jdtoombs) [inline app/editor/src/features/content/form/ContentForm.tsx:90]: Going to ask Bobbi tomorrow where on the layout this should go

**Comment by jdtoombs** [comment]
Putting to draft while I fix some bugs I found

File: app/editor/src/components/tool-bar/styled/ToolBar.tsx [hunk 2]
```diff
@@ -2,7 +2,10 @@ import styled from 'styled-components';
 import { Row } from 'tno-core';
 
 export const ToolBar = styled(Row)`
-  background-color: ${(props) => props.theme.css.backgroundColor};
+  background-color: ${(props) =>
```
Comment (Fosol) [inline app/editor/src/components/tool-bar/styled/ToolBar.tsx:5]: I'm not sure this really makes sense.  If they provide a className the styling for that classname should override the default behaviour on its own.  It shouldn't need logic here.  What you want should be more like a variant option.  Similar to many of our form components.

File: app/editor/src/features/content/form/styled/ContentForm.tsx [hunk 3]
```diff
@@ -20,6 +20,39 @@ export const ContentForm = styled.div`
...
 
+  .src-cpy {
+    border: solid 1px #606060;
+    border-left: none;
+    border-top-right-radius: 0.25em;
+    border-bottom-right-radius: 0.25em;
+  }
+
+  .disabled {
+    color: ${(props) => props.theme.css.lightVariantColor} !important;
```
Comment (Fosol) [inline app/editor/src/features/content/form/styled/ContentForm.tsx:31]: How come we need to use `!important`?  This almost always indicates an implementation issue.

File: app/editor/src/features/content/tool-bar/ContentFormToolBar.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  determineActions: (a: IActionModel) => boolean;
+}
+
+export const ContentFormToolBar: React.FC<IContentFormToolBarProps> = ({
+  contentType,
+  determineActions,
+}) => {
+  const [{ licenses }] = useLookup();
+  const { setFieldValue, values } = useFormikContext<IContentForm>();
+  console.log(licenses.find((l) => l.id === values.licenseId)?.name.replace(/expire/i, ''));
```
Comment (Fosol) [inline app/editor/src/features/content/tool-bar/ContentFormToolBar.tsx:23]: remove

File: app/editor/src/features/content/tool-bar/constants/ActionHelper.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import { ContentTypeName, IActionModel } from 'hooks';
+
+export type ActionHelper = {
+  init?: boolean;
+  determineActions: () => (a: IActionModel) => boolean;
```
Comment (Fosol) [inline app/editor/src/features/content/tool-bar/constants/ActionHelper.ts:5]: I find the `determineActions` property to be hard to understand.

File: app/subscriber/public/keycloak.json [hunk 6]
```diff
@@ -1,6 +1,6 @@
 {
   "realm": "tno",
-  "auth-server-url": "http://localhost:40001/auth/",
+  "auth-server-url": "http://localhost:50001/auth/",
```
Comment (Fosol) [inline app/subscriber/public/keycloak.json:3]: Ha, we'll need to do a lot more work than this to get the subscriber app working.

**Review by Fosol — CHANGES_REQUESTED** [review]

File: app/subscriber/public/keycloak.json (hunk 6 above)
Comment (jdtoombs) [reply app/subscriber/public/keycloak.json:3]: Oops! That was for my local :)

Comment (jdtoombs) [reply app/subscriber/public/keycloak.json:3]: Edited the wrong file lol will remove

File: app/editor/src/components/tool-bar/styled/ToolBar.tsx (hunk 2 above)
Comment (jdtoombs) [reply app/editor/src/components/tool-bar/styled/ToolBar.tsx:5]: Will adjust!

File: app/editor/src/features/content/tool-bar/constants/ActionHelper.ts (hunk 5 above)
Comment (jdtoombs) [reply app/editor/src/features/content/tool-bar/constants/ActionHelper.ts]: Actually an unused file 😛 deleted!

**Review by Fosol — APPROVED** [review]

**Pull request merged**